### PR TITLE
Feature: Implement Assertion 

### PR DESCRIPTION
**Pull request by m2rads**
Add assertion support and DB validation to test framework

Changes:
- Added Jest expect integration for assertions
- Added support for async DB validations in test steps
- Improved error handling and reporting for assertions
- Added test script to verify assertion implementation
- Updated test builder to handle both sync/async test execution

Example usage:

```typescript
.when('Logged in', async () => {
const user = await db.query.users.findFirst();
expect(user).toBeDefined();
})
```

Demo: 
image

### CODE REVIEW

**Comment by m2rads** [comment]
Note: the feature branch feat/github-oauth has been merged into this branch already. Please merge and resolve that branch before merging this one. Thanks

File: README.md [hunk 1]
```diff
@@ -161,6 +161,22 @@ or use the `pnpm` command:
...
+2. Navigate to `Password and Authentication`
+3. Click on `Authenticator App`
+4. Choose `Use your authenticator app`
+5. Click on `Setup key` to grab the OTP secret
+6. Add the OTP secret to your `.env.local` file or use the `shortest` cli to add it:
+
+```bash
+shortest --github-code --secret=<OTP_SECRET>
+```
+7. Laslty enter the 2FA code generated in terminal in your Github Authenticator editor
```
Comment (slavingia) [inline README.md:177]: Typo. Please use AI to rewrite these as I find it does a great job with things like onboarding docs

Comment (m2rads) [reply README.md]: Wrote this section with with AI

**Comment by m2rads** [comment]
Hi @slavingia I addressed your pr review btw. Please let me know if you'd like me to change something else.

**Pull request merged**